src/main/java/com/scmspain/services/TweetService.java

```diff
@@ -10,44 +12,68 @@
 import javax.transaction.Transactional;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 @Service
 @Transactional
 public class TweetService {
     private EntityManager entityManager;
     private MetricWriter metricWriter;
+    private TweetValidationUtils tweetValidationUtils;
+
 
     public TweetService(EntityManager entityManager, MetricWriter metricWriter) {
         this.entityManager = entityManager;
         this.metricWriter = metricWriter;
+        tweetValidationUtils = new TweetValidationUtils();
```

Comment:
Hi @hespoz I see now why do you have doubts about making this static or not static. Why haven't you injected here a TweetValidation instance from the spring context same way you have injected an EntityManager or a MetricWriter.
This way the TweetService is decoupled from the concrete implementation of TweetValidation, and at test time it can be mocked.

Reply:
Hi @rogervinas thats fair, actually I did not think about, but your approach make sense. I always create static classes for Mappers and Validations, because the memory reason I gave last thursday, but for sure I will rethink things hahahahah.

Reply:
@rogervinas regarding the injection of TweetValidationUtils, agree with you, we should leverage the IoC that spring or other frameworks offer to us. Just did not think about it when fix it.